lua/plenary/enum.lua

```diff
@@ -0,0 +1,30 @@
+local Enum = {}
+
+setmetatable(Enum, Enum)
+
+function Enum:__call(tbl)
+  local result = {}
+  local i = 1
+  -- Filter out everything that isn't a string, or a table since we only care
+  -- about those for enum names/values
+  for _, v in ipairs(tbl) do
+    if type(v) == 'string' then
+      result[v] = i
```

Comment:
can you make each variant part of the same metatable so that they can only compare with each other and not other numbers or other enums